src/librustc_parse/lexer/mod.rs

```diff
@@ -380,12 +367,8 @@ impl<'a> StringReader<'a> {
             }
             rustc_lexer::LiteralKind::Float { base, empty_exponent } => {
                 if empty_exponent {
-                    let mut err = self.struct_span_fatal(
-                        start,
-                        self.pos,
-                        "expected at least one digit in exponent",
-                    );
-                    err.emit();
+                    self.fatal_span_(start, self.pos, "expected at least one digit in exponent")
```

Comment:
```suggestion
                    self.sess.span_diagnostic.struct_span_fatal(self.mk_sp(from_pos, to_pos), m).emit();
```
CI fails, so apparently these two don't do the same thing.

Reply:
Yes, it took me some time to figure out what broke.
`struct_span_fatal()` doesn't abort immediately, while `fatal_span_()` does, so fewer error messages are printed in the test cases. I replaced it with `err_span_()`.